### PR TITLE
Feature/#95 set csp headers

### LINKED ISSUE #95: Set CSP headers
https://en.wikipedia.org/wiki/Content_Security_Policy
https://www.owasp.org/index.php/Content_Security_Policy

### PR DESCRIPTION
**Pull request by hixi**
Security enhancement: use csp and serve jsavascript locally only. deny any loading from other sites other than osm.org.

Reviewed by:
- [x] @das-g 

Closes #95

### CODE REVIEW

File: docker_entrypoint/nginx/default.conf [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+# It's usually enabled by default anyway, so the role of this header is to re-enable the filter for
+# this particular website if it was disabled by the user.
+# https://www.owasp.org/index.php/List_of_useful_HTTP_headers
+add_header X-XSS-Protection "1; mode=block";
+
+# with Content Security Policy (CSP) enabled(and a browser that supports it(http://caniuse.com/#feat=contentsecuritypolicy),
+# you can tell the browser that it can only download content from the domains you explicitly allow
+# http://www.html5rocks.com/en/tutorials/security/content-security-policy/
+# https://www.owasp.org/index.php/Content_Security_Policy
+# I need to change our application code so we can increase security by disabling 'unsafe-inline' 'unsafe-eval'
```
Comment (das-g) [inline docker_entrypoint/nginx/default.conf]: Who is "I"?

Comment (hixi) [inline docker_entrypoint/nginx/default.conf]: https://gist.github.com/plentz/6737338

File: docker_entrypoint/nginx/default.conf [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+# directives for css and js(if you have inline css or js, you will need to keep it too).
+# more: http://www.html5rocks.com/en/tutorials/security/content-security-policy/#inline-code-considered-harmful
+add_header Content-Security-Policy "default-src 'self'; script-src 'self'; img-src 'self' http://*.osm.org https://*.osm.org data:; style-src 'self' 'unsafe-inline'; font-src 'self' data:; frame-src; object-src 'none'";
```
Comment (das-g) [inline docker_entrypoint/nginx/default.conf]: Do we use inline scripts or can _we_ disable `'unsafe-inline'`?

Comment (hixi) [inline docker_entrypoint/nginx/default.conf]: inline scripts *are* disabled.

File: osmaxx/excerptexport/templates/excerptexport/base.html [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
     {# MAP SCRIPTS #}
     <script src='{% static "excerptexport/libraries/leaflet/leaflet.js" %}'></script>
-    <script>L.Path.CLIP_PADDING = 1.0;</script>
+    <script src='{% static "excerptexport/scripts/maps/LeafletSetup.js" %}'></script>
```
Comment (das-g) [inline osmaxx/excerptexport/templates/excerptexport/base.html]: :+1:

**Pull request merged**